agent/util/subtree_unix.go

```diff
@@ -145,30 +112,20 @@ func waitForExit(ctx context.Context, pidsToWait []int) ([]int, error) {
 			MaxDelay:    cleanupCheckTimeoutMax,
 		})
 
-	return unkilledPids, err
-}
-
-type process struct {
-	pid     int
-	command string
-	env     []string
+	return unkilledPIDs, err
 }
 
-func psAllProcesses(ctx context.Context) ([]process, error) {
+func psAllProcesses(ctx context.Context) ([]int, error) {
 	/*
-		Usage of ps for extracting environment variables:
-		e: print the environment of the process (VAR1=FOO VAR2=BAR ...)
+		Usage of ps:
 		-A: list *all* processes, not just ones that we own
-		-o: print output according to the given format. We supply 'pid=,command=' to
-		print the pid and command columns without headers
-
-		Each line of output has a format with the pid, command, and environment, e.g.:
-		1084 foo.sh PATH=/usr/bin/sbin TMPDIR=/tmp LOGNAME=xxx
+		-o: print output according to the given format. We supply 'pid=' to
+		print just the pid column without headers.
 	*/
 	psCtx, cancel := context.WithTimeout(ctx, contextTimeout)
 	defer cancel()
 
-	args := []string{"e", "-A", "-o", "pid=,command="}
+	args := []string{"-A", "-o", "pid="}
```

Comment:
for my own understanding, why does using group leaders require that we change ps args in this way? Why do we no longer want the command column?

Reply:
I think since we no longer do it by working directory, since the command column gave us the working directory of the process (jonathan can confirm)

Reply:
I'm with Zack: the command column had been necessary in order to check the working directory.
The other change, to remove the `e` option, is because we no longer check the process's environment variables (which we were unable to get on macOS anyways...)

So basically all we want now is a list of PIDs.

Reply:
makes sense, thank you!